src/pages/Messages.jsx

```diff
@@ -6,34 +6,66 @@ import {
 import React, { useEffect, useState } from "react";
 import MessageBoard from "../components/MessageBoard";
 
-import { firestore } from "../firebase";
+import { auth, firestore } from "../firebase";
 import {
   collection,
   query,
   getDocs,
-  getDoc,
+  where,
   doc,
-  where
+  getDoc,
+  orderBy,
+  limit
 } from "firebase/firestore";
 
 const Messages = () => {
-  const [connectedChats, setConnectedChats] = useState([]);
-  const [childId, setChildId] = useState("");
   const [connectedChatsObjects, setConnectedChatsObjects] = useState([]);
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const collectionRef = collection(firestore, "Child");
-        // get users email from auth ===> auth.email
-        const q = query(collectionRef, where("email", '==', 'penpalprogram.murphycharity@gmail.com'))
-        const docRef = await getDocs(q)
-        if (!docRef.empty) {
-          const connectedChatsRef = docRef.docs[0].data()
-          setChildId(docRef.docs[0].id)
-          console.log("docs 0 ", docRef.docs[0])
-          connectedChatsRef?.connected_chats.forEach(chat => {
-            setConnectedChats([...connectedChats, chat])
-          })
+        if (!auth.currentUser?.uid) {
+          console.warn("error loading auth")
+          setTimeout(() => {
+            fetchData()
+          }, 2000)
+          return
+        }
+        const userDocRef = doc(collection(firestore, "users"), auth.currentUser.uid);
+        const userDocSnapshot = await getDoc(userDocRef);
+
+        if (userDocSnapshot.exists()) {
+          const letterboxQuery = query(collection(firestore, "letterbox"), where("members", "array-contains", userDocRef));
+          const letterboxQuerySnapshot = await getDocs(letterboxQuery);
+
+          const messages = [];
+
+          for (const doc of letterboxQuerySnapshot.docs) {
+            const letterboxData = doc.data();
+            const lettersCollectionRef = collection(doc.ref, "letters");
+            const queryDocumentSnapshots = getDocs(
+              query(lettersCollectionRef,
+                where("deleted_at", "==", null),
+                where("moderation.approved", "==", true),
+                orderBy("deleted_at"),
+                orderBy("moderation.approved"),
+                orderBy("created_at", "desc"),
+                limit(1))
+            )
+            const latestMessage = queryDocumentSnapshots[0].data()
+            console.log(letterboxQuerySnapshot)
```

Comment:
```suggestion
            const lettersQuerySnapshot = await getDocs(
              query(lettersCollectionRef,
                where("deleted_at", "==", null),
                where("moderation.approved", "==", true),
                orderBy("deleted_at"),
                orderBy("moderation.approved"),
                orderBy("created_at", "desc"),
                limit(1))
            )
            const queryDocumentSnapshots = lettersQuerySnapshot.docs
            const latestMessage = queryDocumentSnapshots[0].data()
            console.log(latestMessage)
```

Reply:
Please also make it error-safe like [this](https://github.com/murphy-charitable-foundation/magic-chat-app-web/pull/4/files#diff-c8852b18d7a0b8f4b688392de7ab3082af77bb486e6ed33ccd2807e23ef0f915R69-R71) when the query response returns nothing, but rather it could be an empty screen or "nothing-to-see-here" screen.